src/components/Receive.jsx

```diff
@@ -122,42 +91,11 @@ export default function Receive() {
               </rb.Placeholder>
             )}
             <div className="d-flex justify-content-center" style={{ gap: '1rem' }}>
-              <rb.Button
-                variant="outline-dark"
-                data-bs-toggle="tooltip"
-                data-bs-placement="left"
-                onClick={() => {
-                  copyToClipboard(address, addressCopyFallbackInputRef.current).then(
-                    () => {
-                      setAddressCopiedFlag(addressCopiedFlag + 1)
-                    },
-                    (e) => {
-                      setAlert({ variant: 'warning', message: e.message })
-                    }
-                  )
-                }}
-                disabled={!address || isLoading}
-              >
-                {showAddressCopiedConfirmation ? (
-                  <>
-                    {t('receive.text_copy_address_confirmed')}
-                    <Sprite color="green" symbol="checkmark" className="ms-1" width="20" height="20" />
-                  </>
-                ) : (
-                  t('receive.button_copy_address')
-                )}
-              </rb.Button>
-
-              <input
-                readOnly
-                aria-hidden
-                ref={addressCopyFallbackInputRef}
+              <CopyButtonWithConfirmation
                 value={address}
-                style={{
-                  position: 'absolute',
-                  left: '-9999px',
-                  top: '-9999px',
-                }}
+                text={t('receive.button_copy_address')}
+                successText={t('receive.text_copy_address_confirmed')}
+                disabled={!address || isLoading}
```

Comment:
I don't think this `disabled` prop is being passed further down

Reply:
Ah, thanks – nice catch. Fixed.